caffe2/core/typeid.h

```diff
@@ -1,29 +1,64 @@
-#ifndef CAFFE2_CORE_TYPEID_H_
-#define CAFFE2_CORE_TYPEID_H_
+#pragma once
 
 #include <cassert>
 #include <cstdlib>
 #include <iostream>
-#include <map>
+#include <unordered_map>
 #include <mutex>
 #include <type_traits>
+#include <unordered_set>
 #ifdef __GXX_RTTI
-#include <set>
 #include <typeinfo>
 #endif
 
 #include <exception>
 
 #include "caffe2/core/common.h"
+#include "caffe2/utils/IdWrapper.h"
 
 namespace caffe2 {
 
-typedef intptr_t CaffeTypeId;
-std::map<CaffeTypeId, string>& gTypeNames();
-std::set<string>& gRegisteredTypeNames();
+/**
+ * Dynamic type ID of a Tensor argument.  It represents something like CPUFloatTensor, etc.
```

Comment:
is this comment invalid? CPUFloatTensor according to agreed design (and c10 prototype) is not a concrete class, but rather CPUTensorImpl with another TypeId inside. Is my understanding correct?

If so - let's clarify it here

Reply:
Yes, this is invalid. Good catch. It's rather a general type id, used for dtypes, not for tensor types like CPUTensor.